### ISSUE TITLE
Singleton lock issue

### ISSUE THREAD

**Issue by iremmats** [Azure/azure-webjobs-sdk#634]
We are running Kudu locally since we are restricted ip wise. 

One of our continuous jobs has a ServiceBusTrigger function which seems to sometimes pick up messages more than once (loosing lock of msg and next job pick up another). Effect is that we are creating duplicate messages. After looking at the logs in storage I find this error around the times where duplicates are created. 

"Unable to acquire singleton lock blob lease for blob ... (timeout of 0:01:00 exceeded)"
Am also able to find this in .params file
{
  "(singleton)": {
    "Type": "Singleton",
    "TimeToAcquireLock": "00:00:54.0340959",
    "LockOwner": "d23aa663-93ac-4efd-a0a4-d66a2690980f",
    "LockAcquired": false
  }
}

This has so far only happened during batch runs and not during normal traffic. 

In plain english... what does this mean? :)

Are we pushing to much traffic at the same time? Can we prevent the duplicate messages from arising by just increasing the timeout? 

I also find this
{
  "(singleton)": {
    "Type": "Singleton",
    "TimeToAcquireLock": "00:00:03.0111830",
    "LockAcquired": false
  }
}
What does that mean for the behavior of the job?

**Comment by mathewc** [comment]
Uh, I read that a couple times, but I still don't understand :) Can you provide more details or specific repro steps?

**Comment by christopheranderson** [comment]
Closing this since there isn't any recent activity. Message back if you want us to reopen. :)
